### DISCUSSION TITLE
Event Binding with TableView

### DISCUSSION THREAD

**Discussion by TinySkillet** [israel-dryer/ttkbootstrap#514] [Q&A] [answered]
Is there a way to bind a selection event to the Tableview widget? I want to get the contents of selected rows and tried using the << TreeviewSelect >> but it did not function.

**Comment by ghost** [comment]
You have to bind to the underlying 'view'.

For a double click you can bind to `<Double-1>`  for example:

`self.my_tableview.view.bind("<Double-1>", self.row_selected)`

For a simple selection try:

`self.my_tableview.view.bind("<<TreeviewSelect>>", self.row_selected)`

with a method along the lines of:

```
def row_selected(self, event: Any) -> None:
        self._selected_rows: list[Any] = self.my_tableview.get_rows(selected=True)
```

(I'm using Any because I am not sure of the datatypes.)

**Comment by israel-dryer** [comment, marked as answer]
The earlier answer here is correct — bind to the underlying `.view` (the `ttk.Treeview`), not the `Tableview` frame. `<<TreeviewSelect>>` is emitted by the Treeview, which is why binding it to the `Tableview` did nothing:

```python
my_tableview.view.bind("<<TreeviewSelect>>", on_select)   # or "<Double-1>"

def on_select(event):
    rows = my_tableview.get_rows(selected=True)   # list[TableRow]
    data = [r.values for r in rows]               # the actual cell values
    print(data)
```

`get_rows(selected=True)` returns `TableRow` objects for the currently-selected rows; call `.values` on each to get the row's cell data.

See also #347, which covers the same double-click-to-read-row question.
